pkg/controller/eventing/inmemory/channel/reconcile.go

```diff
@@ -203,16 +206,23 @@ func (r *reconciler) createNewConfigMap(data map[string]string) *corev1.ConfigMa
 	}
 }
 
-func multiChannelFanoutConfig(channels []eventingv1alpha1.Channel) *multichannelfanout.Config {
+func multiChannelFanoutConfig(channels []eventingv1alpha1.Channel, logger *zap.Logger) *multichannelfanout.Config {
 	cc := make([]multichannelfanout.ChannelConfig, 0)
 	for _, c := range channels {
 		channelConfig := multichannelfanout.ChannelConfig{
 			Namespace: c.Namespace,
 			Name:      c.Name,
 		}
 		if c.Spec.Subscribable != nil {
+			// TODO After in-memory-channel is retired, this logic must be refactored.
+			asyncHandler := false
+			if c.Spec.Provisioner.Name == asyncProvisionerName {
```

Comment:
Add a unit test to verify the state of `asyncHandler`.

Reply:
will do.